File: template/js/v1/state.logic.js
// version: V1

const util = require("util")
const {{=it.StateName}} = function (context) {
	this.context = context
}

// Partial evaluate
require("./{{=it.StateName}}.evaluate")({{=it.StateName}})

// ----------
// State Logic Implement Below
// ----------

{{=it.StateName}}.prototype.OnUpdate = function(stateData, dt) {	
	// Logic Code Here

	return this._evaluate(stateData)
};

{{=it.StateName}}.prototype.OnEnter = function(stateData) {
	// Logic Code Here
	console.log("{{=it.StateName}} - Enter")
};

{{=it.StateName}}.prototype.OnExit = function(stateData) {
	// Logic Code Here
	console.log("{{=it.StateName}} - Exit")
};

{{=it.StateName}}.prototype.OnEvent = function(stateData, args) {
	// Logic Code Here
	console.log("{{=it.StateName}} - On Event ", args)
};

module.exports = {{=it.StateName}}

File: template/cs/v1/Logic/StateLogic.cs
using System;
using FsmCodeBase;

// ----------
// State Logic Implement Below
// ----------

namespace Fsm.Generated.{{=it.FsmName}}
{
    partial class {{=it.StateName}} : BaseState
    {
        public override string GetName()
        {
            return "{{=it.StateName}}";
        }

        public override void OnEnter(StateMachine context)
        {
            // Logic Code Here
            base.OnEnter(context);
        }

        public override string OnUpdate(float dt, StateMachine context)
        {
            // Logic Code Here
            return base.OnUpdate(dt, context);
        }

        public override void OnEvent(string name, object args, StateMachine context)
        {
            // Logic Code Here
            base.OnEvent(name, args, context);
        }

        public override void OnExit(StateMachine context)
        {
            // Logic Code Here
            base.OnExit(context);
        }
    }
}
